src/transforms/filter.js

```diff
@@ -190,85 +187,71 @@ function getFilterFunc(opts, d2c) {
-                    return cv > coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv > coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv < coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv >= coercedValue[0] && cv <= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'notwithin':
-
-            if(coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv < coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv > coercedValue[1];
-                };
-            }
-            else if(!coercedStrict[0] && !coercedStrict[1]) {
-                return function(v) {
-                    var cv = d2c(v);
-                    return cv <= coercedValue[0] || cv >= coercedValue[1];
-                };
-            }
-
-            break;
-
-        case 'in':
+            return function(v) { return d2c(v) > coercedValue; };
+
+        case '>=':
+            return function(v) { return d2c(v) >= coercedValue; };
+
+        case '[]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '()':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '[)':
+            return function(v) {
+                var cv = d2c(v);
+                return cv >= coercedValue[0] && cv < coercedValue[1];
+            };
+
+        case '(]':
+            return function(v) {
+                var cv = d2c(v);
+                return cv > coercedValue[0] && cv <= coercedValue[1];
+            };
+
+        case '][':
+            return function(v) {
+                var cv = d2c(v);
+                return cv < coercedValue[0] || cv > coercedValue[1];
```

Comment:
@alexcjohnson just to clarify myself, when I wrote 'complement' it's meant it in the set operations sense, for example, the complement of `[8, 12]` is the union of `(-∞, 8)` and `(12, ∞)` which, at least according to my comment you quote, from the wiki, can be denoted as `)8, 12(` though again, whichever notation you guys settle will work out just fine.